### PR TITLE
Miscellaneous facility-based report fixes

### PR DESCRIPTION
**Pull request by gcapalbo**
Some miscellaneous things I ran into while trying to use the new facility-based reports. Best reviewed commit by commit
@czue

### CODE REVIEW

File: logistics_project/apps/malawi/warehouse/report_utils.py [hunk 1]
```diff
@@ -214,17 +214,16 @@ def get_reporting_rates_chart(location, start, end, is_facility=False):
...
     Reporting rates chart for multiple facilities, for a single 
     month.
     """
     report_chart = _reporting_rate_chart_defaults()
     data = defaultdict(lambda: defaultdict(lambda: 0)) # turtles!
-    rrs = ReportingRate.objects.filter(supply_point__in=supply_points, date=date)
+    rrs = ReportingRate.objects.filter(supply_point__in=supply_points, date=date, is_facility=is_facility)
     for rr in rrs:
         sp = rr.supply_point
-        rr = ReportingRate.objects.get(supply_point=sp, date=date)
```
Comment (gcapalbo) [inline logistics_project/apps/malawi/warehouse/report_utils.py]: as far as i can tell this line is redundant. i also opened the dashboard with and without this line and it looked the same

Comment (czue) [inline logistics_project/apps/malawi/warehouse/report_utils.py]: yeah, this just looks like a bad refactor introduced [here](https://github.com/dimagi/logistics/commit/2b80c8a4ca15c0013b67d2874bc5c5b0b2ce8615#diff-1a28246a977acc7ec4d6333781b34b81R213). thanks!

File: logistics_project/apps/malawi/warehouse/report_views/dashboard.py [hunk 2]
```diff
@@ -68,7 +68,13 @@ def custom_context(self, request):
...
-                "graphdata": get_multiple_reporting_rates_chart(child_sps, window_date)}
\ No newline at end of file
+        return {
+            "window_date": window_date,
+            "dsummary_table": dsummary_table,
+            "alert_table": alert_table,
+            "graphdata": get_multiple_reporting_rates_chart(
+                child_sps,
+                window_date,
+                is_facility=request.is_facility
```
Comment (czue) [inline logistics_project/apps/malawi/warehouse/report_views/dashboard.py]: thanks for the fix. i must have just missed this.

File: logistics_project/apps/malawi/warehouse/warehouse_view.py [hunk 3]
```diff
@@ -16,6 +18,27 @@ class MalawiWarehouseView(ReportView):
...
+    # If overridden to True, implement get_min_start_date() and the start date
+    # for the report will automatically update if the default range or the
+    # user-selected range is too wide.
+    automatically_adjust_datespan = False
+
+    def get_min_start_date(self, request):
+        """
+        Should return the minimum start date that should be used with this report.
+        """
+        raise NotImplementedError()
```
Comment (czue) [inline logistics_project/apps/malawi/warehouse/warehouse_view.py]: does every report need to implement this? alternative would be to just default return `datetime.min` which might be an ok default.

File: logistics_project/apps/malawi/warehouse/warehouse_view.py [hunk 4]
```diff
@@ -16,6 +18,27 @@ class MalawiWarehouseView(ReportView):
...
+
+    def update_datespan(self, request):
+        min_date = self.get_min_start_date(request)
+        if min_date and min_date > request.datespan.startdate:
+            startdate = min_date
+            enddate = request.datespan.enddate
+            if enddate < startdate:
+                enddate = startdate
+            request.datespan = DateSpan(startdate, enddate, request.datespan.format)
+            messages.warning(request, "The date range has been automatically adjusted because it was too wide.")
```
Comment (czue) [inline logistics_project/apps/malawi/warehouse/warehouse_view.py]: nit: "too wide" might be a little confusing. maybe "The date range has been automatically adjusted to fit the oldest available data" or something?

File: logistics_project/apps/malawi/warehouse/runner.py [hunk 5]
```diff
@@ -55,6 +55,7 @@ class MalawiWarehouseRunner(WarehouseRunner):
     skip_historical_stock = False
     consumption_test_mode = False
     hsa_limit = 0
+    facility_limit = 0
```
Comment (czue) [inline logistics_project/apps/malawi/warehouse/runner.py]: whoops sorry this must have been a last minute pre-commit omission

File: logistics_project/apps/malawi/warehouse/warehouse_view.py (hunk 3 above)
Comment (gcapalbo) [inline logistics_project/apps/malawi/warehouse/warehouse_view.py]: only reports that override `automatically_adjust_datespan = True` need to implement it - i added a comment above the `automatically_adjust_datespan` property but can add one here too

File: logistics_project/apps/malawi/warehouse/warehouse_view.py (hunk 4 above)
Comment (gcapalbo) [inline logistics_project/apps/malawi/warehouse/warehouse_view.py]: yeah that sounds better, will update

**Pull request merged**